common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -964,4 +969,19 @@ private void saveMsaAccountToCache(final @NonNull Bundle resultBundle,
     private void msalOAuth2TokenCacheSetSingleSignOnState(@SuppressWarnings(WarningType.rawtype_warning) @NonNull MsalOAuth2TokenCache msalOAuth2TokenCache, MicrosoftStsAccount microsoftStsAccount, MicrosoftRefreshToken microsoftRefreshToken) throws ClientException {
         msalOAuth2TokenCache.setSingleSignOnState(microsoftStsAccount, microsoftRefreshToken);
     }
+
+    /**
+     * Verifies if the token parameters are supported by the required broker protocol version
+     * @param parameters Token Parameters for verify
+     * @throws ClientException if the token parameters are not supported
+     */
+    private void verifyTokenParametersAreSupported(@NonNull final TokenCommandParameters parameters) throws ClientException {
+        final String requiredProtocolVersion = parameters.getRequiredBrokerProtocolVersion();
+        if (parameters.getAuthenticationScheme() instanceof PopAuthenticationSchemeWithClientKeyInternal
+                && !BrokerProtocolVersionUtil.canSupportPopAuthenticationSchemeWithClientKey(requiredProtocolVersion)){
+            throw new ClientException(ClientException.AUTH_SCHEME_MISMATCH,
```

Comment:
AUTH_SCHEME_MISMATCH would be used when response from server does not match requested scheme right? Here I think we are verifying if the scheme requested is supported by current broker version. So should we use a different client exception like AUTH_SCHEME_NOT_SUPPORTED?

Reply:
+1 to Sowmya's comment. Let's use a more clearer and specific error code here